src/masternode.h

```diff
@@ -258,6 +259,19 @@ class CMasternode
 
     bool IsWatchdogExpired() { return nActiveState == MASTERNODE_WATCHDOG_EXPIRED; }
 
+    bool IsValidForPayment()
+    {
+        if(nActiveState == MASTERNODE_ENABLED) {
+            return true;
+        }
+        if(!sporkManager.IsSporkActive(SPORK_14_REQUIRE_SENTINEL_FLAG) &&
```

Comment:
Looks like this is broken:
`sporkManager.IsSporkActive(SPORK_14_REQUIRE_SENTINEL_FLAG) == true` -> `IsValidForPayment() == false` regardless of `nActiveState`

Also the common pattern for bool functions in current code is to fail early on some conditions and to return true if nothing fails, not the otherwise, so I propose to rewrite IsValidForPayment() to smth like:
```
bool IsValidForPayment()
{
    if(IsWatchdogExpired()) {
        return !sporkManager.IsSporkActive(SPORK_14_REQUIRE_SENTINEL_FLAG);
    }
    return IsEnabled();
}
```
which can be further shrunk to one-liner like `return (IsWatchdogExpired() && !sporkManager.IsSporkActive(SPORK_14_REQUIRE_SENTINEL_FLAG)) || IsEnabled();` but I guess the first one is a bit easier to read :)